main.tf

```diff
@@ -34,7 +34,7 @@ resource "aws_db_instance" "this" {
   replica_mode                          = var.replica_mode
   db_name                               = var.engine == try("mysql", "posgresql") ? var.name : null
   option_group_name                     = length(var.option_group_name) > 0 ? var.option_group_name : join("", aws_db_option_group.this.*.name)
-  parameter_group_name                  = var.parameter_group_name
+  parameter_group_name                  = length(var.parameter_group) > 0 ? join("", aws_db_parameter_group.this.*.id) : var.parameter_group_name
```

Comment:
The creation of the option_group is defined by a flag, which is not present for the parameter_group. The other comment talks about this more in depth.

Reply:
Addressed @mariomerco